src/main/JayZ.ts

```diff
@@ -107,6 +103,30 @@ export class JayZ {
     return Promise.all(itemPromises)
   }
 
+  private generateDecryptParams<T, K extends keyof T>(
```

Comment:
This is a stylistic thing, but if it were me, I'd just inline this function above on line 81 as the signature of your function ends up being almost as long as the function body itself. 

One other thought here is that TypeScript's [type guards](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards) are useful for this sort of use-case where you want to check if a type is an A or a B and have TS infer narrow the type to A or B (i.e legacy or contemporary formats).